### PR TITLE
Limit retries on InvalidSessionException.

### PR DESCRIPTION
**Pull request by aurghob33**
## Description

Change avoids an infinite loop by placing a retry limit on an ISE.
Also bumped version as part of this change.
CHANGELOG updated.

## Motivation and Context

In the earlier version of the driver, an infinite loop would have resulted on an ISE.
We want to limit the possibility of an infinite loop in the driver, especially on an ISE.

## Testing

Added Unit test specific to the change added.
Ran:
mvn -B package --file pom.xml
mvn verify -Dregion=us-east-1

Both integration tests and unit tests ran successfully.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Checklist

- [x] Local run of `mvn install` succeeds
- [x] My code follows the code style of this project
- [ ] My change requires a change to the Javadoc documentation
- [ ] I have updated the Javadoc documentation accordingly
- [x] I have read the **README** document
- [x] I have added tests to cover my changes
- [x] All new and existing tests passed
- [x] A short description of the change has been added to the **CHANGELOG**

## License

- [x] I confirm that this pull request can be released under the Apache 2 license

### CODE REVIEW

File: src/main/java/software/amazon/qldb/QldbDriverImpl.java [hunk 1]
```diff
@@ -94,6 +95,12 @@ protected QldbDriverImpl(String ledgerName,
         this.isClosed = new AtomicBoolean(false);
         this.readAhead = readAhead;
         this.executorService = executorService;
+        int iseRetryLimit = maxConcurrentTransactions + 3;
+        if (maxConcurrentTransactions > Integer.MAX_VALUE - 3) {
```
Comment (simonz-bq) [inline src/main/java/software/amazon/qldb/QldbDriverImpl.java:99]: Wouldn't it be better to do the check before assigning maxConccurentTransactions +3? 

If going over the max value in l98 isn't problematic then it's fine, but just to be safe.

File: src/main/java/software/amazon/qldb/QldbDriverImpl.java [hunk 2]
```diff
@@ -134,12 +141,18 @@ public void execute(ExecutorNoReturn executor, RetryPolicy retryPolicy) {
         }
 
         ExecutionContext executionContext = new ExecutionContext();
+        int getSessionAttempt = 0;
         while (true) {
             QldbSession qldbSession = null;
             try {
+                getSessionAttempt += 1;
```
Comment (simonz-bq) [inline src/main/java/software/amazon/qldb/QldbDriverImpl.java:148]: This variable is named poorly. qldbSession.execute could be the source of the ISE, not only getSession.

File: src/test/software/amazon/qldb/QldbDriverImplTest.java [hunk 3]
```diff
@@ -312,6 +317,74 @@ public void testExecuteStatementOccConflict() throws IOException {
...
+                    .message(iseMessage)
+                    .build());
+        }
+
+        Executor<Boolean> executor = txn -> {
+            txn.execute(statement, Collections.emptyList());
+            return true;
+        };
+
+        Session mockSession = Session.startSession(LEDGER, mockClient);
```
Comment (byronlin-bq) [inline src/test/software/amazon/qldb/QldbDriverImplTest.java:350]:
```suggestion
        Session mockSession1 = Session.startSession(LEDGER, mockClient);
```

Comment (aurghob33) [reply src/test/software/amazon/qldb/QldbDriverImplTest.java:350]: Addressed - will update

File: src/main/java/software/amazon/qldb/QldbDriverImpl.java (hunk 1 above)
Comment (aurghob33) [reply src/main/java/software/amazon/qldb/QldbDriverImpl.java:99]: Just a style preference - keeping as is.

File: src/main/java/software/amazon/qldb/QldbDriverImpl.java (hunk 2 above)
Comment (aurghob33) [reply src/main/java/software/amazon/qldb/QldbDriverImpl.java:148]: Renaming to iseRetryAttempt and moving the increment down.

**Pull request merged**